2006-12-05  Jakub Jelinek  <jakub@redhat.com>

	* nis/nis_subr.c (nis_getnames): Revert last change.

2006-10-11  Jakub Jelinek  <jakub@redhat.com>

	* nis/nis_defaults.c (__nis_default_access): Don't call getenv twice.

	* nis/nis_subr.c (nis_getnames): Use __secure_getenv instead of getenv.
	* sysdeps/generic/unsecvars.h: Add NIS_PATH.

--- libc/nis/nis_defaults.c	10 May 2006 02:54:46 -0000	1.8
+++ libc/nis/nis_defaults.c	11 Oct 2006 16:22:34 -0000	1.9
@@ -447,7 +447,7 @@ __nis_default_access (char *param, unsig
     {
       cptr = getenv ("NIS_DEFAULTS");
       if (cptr != NULL && strstr (cptr, "access=") != NULL)
-	result = searchaccess (getenv ("NIS_DEFAULTS"), result);
+	result = searchaccess (cptr, result);
     }
 
   return result;
--- libc/sysdeps/generic/unsecvars.h	6 Jan 2005 22:40:19 -0000	1.8
+++ libc/sysdeps/generic/unsecvars.h	11 Oct 2006 16:24:05 -0000	1.9
@@ -18,6 +18,7 @@
   "LOCALDOMAIN\0"							      \
   "LOCPATH\0"								      \
   "MALLOC_TRACE\0"							      \
+  "NIS_PATH\0"								      \
   "NLSPATH\0"								      \
   "RESOLV_HOST_CONF\0"							      \
   "RES_OPTIONS\0"							      \
